Source/MLX/Random.swift

```diff
@@ -3,6 +3,70 @@
 import Cmlx
 import Foundation
 
+/// Collection of functions related to random number generation.
+///
+/// Following [JAX’s PRNG design](https://jax.readthedocs.io/en/latest/jep/263-prng.html)
+/// we use a splittable version of Threefry, which is a counter-based PRNG.
+///
+/// Random sampling functions in MLX use an implicit global PRNG state by default.
+/// However, all functions take an optional key keyword argument for when more fine-grained
+/// control or explicit state management is needed.  Callers can also arrange for `Task` local
+/// random state -- useful in multithreaded situations.
+///
+/// For example, you can generate random numbers with:
+///
+/// ```swift
+/// for _ in 0 ..< 3 {
+///   print(MLXRandom.uniform())
+/// }
+/// ```
+///
+/// which will print a sequence of unique pseudo random numbers. Alternatively you can explicitly set the key:
+///
+/// ```swift
+/// let key = MLXRandom.key(0)
+/// for _ in 0 ..< 3 {
+///   print(MLXRandom.uniform(key: key))
+/// }
+/// ```
+///
+/// which will yield the same pseudo random number at each iteration as the key doesn't change.
+///
+/// You can also use a ``RandomState`` to generate different random numbers but the same sequence
+/// each time:
+///
+/// ```swift
+/// let state = RandomState(seed: 0)
+/// for _ in 0 ..< 3 {
+///   print(MLXRandom.uniform(key: state))
+/// }
```

Comment:
I think it would be helpful in the exposition here to add an example inbetween this one and the previous one that explains how to split the array random key. So concretely:

1. Example with array key -> same numbers
2. Example with array key + split -> different numbers
3. Same thing as 2 but nicer syntax with random state